From 1ab5feb6260589f254ed209816cb67dbe9d3e4a5 Mon Sep 17 00:00:00 2001
From: =?utf-8?q?Trond=20Kjern=C3=A5sen?= <trond@trolltech.com>
Date: Mon, 1 Mar 2010 13:44:22 +0100
Subject: [PATCH] Fixed QPixmap::load() to not modify referenced copies (again!)

Change 8721d060a67a01ac891cab9d3d17aacf7373bcf0 broke the previous
fix.

Task-number: QTBUG-8606
Reviewed-by: Gunnar
---
 src/gui/image/qpixmap.cpp |   19 ++++++-------------
 1 files changed, 6 insertions(+), 13 deletions(-)

--- a/src/gui/image/qpixmap.cpp
+++ b/src/gui/image/qpixmap.cpp
@@ -831,21 +831,14 @@ bool QPixmap::load(const QString &fileNa
     if (QPixmapCache::find(key, *this))
         return true;
 
-    bool ok;
-
-    if (data) {
-        ok = data->fromFile(fileName, format, flags);
-    } else {
-        QScopedPointer<QPixmapData> tmp(QPixmapData::create(0, 0, QPixmapData::PixmapType));
-        ok = tmp->fromFile(fileName, format, flags);
-        if (ok)
-            data = tmp.take();
-    }
-
-    if (ok)
+    QScopedPointer<QPixmapData> tmp(QPixmapData::create(0, 0, data ? data->type : QPixmapData::PixmapType));
+    if (tmp->fromFile(fileName, format, flags)) {
+        data = tmp.take();
         QPixmapCache::insert(key, *this);
+        return true;
+    }
 
-    return ok;
+    return false;
 }
 
 /*!
